Sources/NIOSOCKS/Messages/ServerResponse.swift

```diff
@@ -0,0 +1,114 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+// MARK: - ServerResponse
+
+/// The SOCKS Server's response to the client's request
+/// indicating if the request succeeded or failed.
+public struct ServerResponse: Hashable {
+    
+    /// The SOCKS protocol version - we currently only support v5.
+    public var version: UInt8
+    
+    /// The status of the connection - used to check if the request
+    /// succeeded or failed.
+    public var reply: Reply
+    
+    /// The host address.
+    public var boundAddress: AddressType
+    
+    /// The host port.
+    public var boundPort: UInt16
+    
+    /// Creates a new `ServerResponse`.
+    /// - parameter reply: The status of the connection - used to check if the request
+    /// succeeded or failed.
+    /// - parameter boundAddress: The host address.
+    /// - parameter boundPort: The host port.
+    public init(reply: Reply, boundAddress: AddressType, boundPort: UInt16) {
+        self.version = 5
+        self.reply = reply
+        self.boundAddress = boundAddress
+        self.boundPort = boundPort
+    }
+    
+    init?(buffer: inout ByteBuffer) {
+        guard
+            let version = buffer.readInteger(as: UInt8.self),
+            let reply = Reply(buffer: &buffer),
+            buffer.readBytes(length: 1) == [0x00],
+            let boundAddress = AddressType(buffer: &buffer),
+            let boundPort = buffer.readInteger(as: UInt16.self)
+        else {
+            return nil
+        }
+        self.version = version
+        self.reply = reply
+        self.boundAddress = boundAddress
+        self.boundPort = boundPort
+    }
+}
+
+// MARK: - Reply
+
+/// Used to indicate if the SOCKS client's connection request succeeded
+/// or failed.
+public struct Reply: Hashable {
+    
+    /// The connection succeeded and data can now be transmitted.
+    static var succeeded = Reply(value: 0x00)
```

Comment:
These all need to be `static let`: we have no reason to allow anyone to modify them at runtime.